flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleEnvironment.java

```diff
@@ -92,25 +92,34 @@
 	 */
 	int start() throws IOException;
 
+	/**
+	 * Create a context of the shuffle input/output owner used to create partitions or gates belonging to the owner.
+	 *
+	 * <p>This method has to be called only once to avoid duplicated internal metric group registration.
+	 *
+	 * @param ownerName the owner name, used for logs
+	 * @param executionAttemptID execution attempt id of the producer or consumer
```

Comment:
of the producer or consumer -> of the owner?

Reply:
At the moment we have exec id only for tasks/executions, in this context producer or consumer. I thought about owner as a potentially any shuffle IO component within task/execution, possibly even source/sink